src/Symfony/Bundle/FrameworkBundle/Client.php

```diff
@@ -104,6 +106,45 @@ public function enableReboot()
         $this->reboot = true;
     }
 
+    /**
+     * Logins a user against a specific firewall.
+     *
+     * @param string|object $user
+     * @param string        $credentials
+     * @param string[]      $roles
+     * @param string        $firewallName
+     * @param string|null   $firewallContext
+     */
+    public function login(
+        $user,
+        string $credentials,
+        array $roles = array('ROLE_ADMIN'),
```

Comment:
granting an ADMIN role by default looks wrong to me.

Reply:
and in case the user is an object implementing UserInterface, I suggest taking roles from the object instead.

Reply:
I'd even suggest to make it an empty array by default and merge it in user's role if populated, WDYT?

Reply:
👍